tests/array_table_model_tester.ts

```diff
@@ -36,4 +51,165 @@ export class ArrayTableModelTester {
     Expect(receivedIndex).toEqual(1);
     listener.unlisten();
   }
+  /** Tests removing rows. */
+  @Test()
+  public testRemoveRow(): void {
+    const model = new ArrayTableModel();
+    let receivedIndex = undefined;
+    let removedRow = new ArrayTableModel();
+    const slot = (operations: Operation[]) => {
+      if(operations) {
+        const operation = operations[operations.length - 1];
+        if(operation instanceof RemoveRowOperation ) {
+          receivedIndex = operation.index;
+          if(operation.row instanceof ArrayTableModel) {
+            removedRow = operation.row;
+          }
+        } else if (operation instanceof AddRowOperation) {
+          return;
+        } else {
+          Expect(false).toEqual(true);
+        }
+      } else {
+        Expect(false).toEqual(true);
+      }
+    };
+    const listener = model.connect(slot);
+    Expect(model.rowCount).toEqual(0);
```

Comment:
You don't need to duplicate all these tests. If this has been tested in another unit test then remove it from this one.